kong/plugins/opentelemetry/handler.lua

```diff
@@ -78,17 +78,28 @@ local function http_export_request(conf, pb_data, headers)
   return true
 end
 
+local prev_endpoint = nil
+local parsed_endpoint = nil
+local ENV_IDENTIFIER_REGEX = '%${([%w-_]+)}'
+
 local function http_export(conf, spans)
   local start = ngx_now()
   local headers = get_cached_headers(conf.headers)
   local payload = encode_traces(spans, conf.resource_attributes)
 
-  local ok, err = http_export_request(conf, payload, headers)
+  -- caching because comparing string is much faster than applying regex
+  if conf.endpoint ~= prev_endpoint then
+    prev_endpoint = conf.endpoint
+    parsed_endpoint = string.gsub(conf.endpoint, ENV_IDENTIFIER_REGEX, os.getenv)
```

Comment:
unfortunately env vars are not available to read once Kong is started, so we can't really
have dynamic env var names. it can only be static like datadog plugin does.

Reply:
Yes, That's similar to any app, right? We need to set the environment variable before starting the process. Here, user will set the environment variables upfront in Kong's helm values.yaml with following syntax, and refer to them in the plugin conf
```
env:
  NGINX_MAIN_ENV: "KONG_HOST_IP"
  KONG_HOST_IP: 
      valueFrom:
      fieldRef:
        apiVersion: v1
        fieldPath: status.hostIP
```

Reply:
The problem here is nginx clears all environment variable after it's restarted (i.e. post the `init` phase). In regards to lua code, that means only `os.getenv` that's called at module level will return exepcted result. The approach in this PR that reads env var in a function, that's called during run-time, will not work.

Reply:
>  The approach in this PR that reads env var in a function, that's called during run-time, will not work.

@fffonion  I tested this PR plugin and  It's already working. It's runtime, and it's able to pick up the environment variable, `NGINX_MAIN_ENV` directive is to pass the environment variables to kong, right? 

What do you mean after the `restart`, it won't work?  Can you give me a scenario to reproduce that?

Reply:
That way it would work yes. But this also makes the design more hackier.
We appreciate your continous effort on pushing this forward, but I'm afraid these are not the propriate approach we feel would solve the issue properly as the reason I explained in https://github.com/Kong/kong/pull/9466#issuecomment-1451302916.

On the other side, if you would like to give a stab on making the datadog and otel plugin able to be use with secrets management, your contributions are always welcomed.

Reply:
ping @fffonion

Reply:
@jaswanthikolla right, that wasn't the right/supported syntax for referencable values, consider using a different approach that does the templating at helm chart side and pass in the whole string in.